### ISSUE TITLE
Uncaught TypeError: _selectedDate.isSame is not a function(…)

### ISSUE THREAD

**Issue by Stas-Buzunko** [clauderic/react-infinite-calendar#61]
Hi!
Config 
```
<InfiniteCalendar afterSelect={onDateChange} selectedDate={false} disabledDates={bookedDates} minDate={moment().startOf('day')} />
```
Even on outside calendar click, this validation is called. And because selectedDate={false} it produces an error. The question is why componentsWillReceiveProps is called on outside click
![image](https://cloud.githubusercontent.com/assets/12331100/20702509/3333bad0-b629-11e6-801c-2926bc06f4e9.png)

**Comment by Stas-Buzunko** [comment]
Clicks on dates work properly

**Comment by clauderic** [comment]
Hey @Stas-Buzunko, thanks for reporting this.

I believe this is related to #63, though just to be sure, would you be able to put together a jsfiddle to replicate this issue? You can fork this fiddle as a starting point: https://jsfiddle.net/clauderic/4rhn03do/

**Comment by clauderic** [comment]
Hey Stas-Buzunko, FYI, I just shipped version 2.0.0, which replaces moment.js in favor of date-fns. This should no longer be an issue, though let me know if you run into any other issues
